{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#Gamak Pond Sequencing Results \n",
      "\n",
      " Sequence:  GamakSite1Raw_58480 \n",
      " Accuracy:  86.985 %\n",
      " Classify:  4463866\n",
      "              k__Bacteria;\n",
      "              p__Bacteroidetes;\n",
      "              c__Bacteroidia;\n",
      "              o__Bacteroidales;\n",
      "              f__Porphyromonadaceae;\n",
      "              g__;\n",
      "              s__ \n",
      "\n",
      " Sequence:  GamakSite2Raw_102642 \n",
      " Accuracy:  90.281 %\n",
      " Classify:  4463866\n",
      "              k__Bacteria;\n",
      "              p__Bacteroidetes;\n",
      "              c__Bacteroidia;\n",
      "              o__Bacteroidales;\n",
      "              f__Porphyromonadaceae;\n",
      "              g__;\n",
      "              s__ \n",
      "\n",
      " Sequence:  GamakSite2Raw_39621 \n",
      " Accuracy:  92.763 %\n",
      " Classify:  354006\n",
      "              k__Bacteria;\n",
      "              p__Proteobacteria;\n",
      "              c__Alphaproteobacteria;\n",
      "              o__Sphingomonadales;\n",
      "              f__Erythrobacteraceae;\n",
      "              g__Erythrobacter;\n",
      "              s__ \n",
      "\n",
      " Sequence:  GamakSite1Raw_91546 \n",
      " Accuracy:  91.228 %\n",
      " Classify:  354006\n",
      "              k__Bacteria;\n",
      "              p__Proteobacteria;\n",
      "              c__Alphaproteobacteria;\n",
      "              o__Sphingomonadales;\n",
      "              f__Erythrobacteraceae;\n",
      "              g__Erythrobacter;\n",
      "              s__ \n",
      "\n",
      "\n",
      " \n",
      " #Human and Fly ensulin compare \n",
      "\n",
      "Query Acc.ver:         F33H1.1b\n",
      "Subject Acc.ver:       RFX3|ENSP00000371434\n",
      "% Identity:            36.767\n",
      "Alignment Length:      563\n",
      "Number of Mismatches:  270\n",
      "Number of Gap-opens:   14\n",
      "Query Start:           261\n",
      "Query End:             798\n",
      "Subject Start:         184\n",
      "Subject End:           685\n",
      "E-value:               2.55e-107\n",
      "Bit Score:             345 \n",
      "\n",
      "Query Acc.ver:         FBpp0297139\n",
      "Subject Acc.ver:       RFX3|ENSP00000371434\n",
      "% Identity:            56.619\n",
      "Alignment Length:      491\n",
      "Number of Mismatches:  184\n",
      "Number of Gap-opens:   8\n",
      "Query Start:           576\n",
      "Query End:             1055\n",
      "Subject Start:         184\n",
      "Subject End:           656\n",
      "E-value:               0.0\n",
      "Bit Score:             550 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "gamakseq = open('Gamak_result.txt','r')\n",
    "taxonomy = open('GreenGene_135.79_otus_taxonomy.txt','r')\n",
    "humanflyworm = open('HumanFlyWorm_result.txt','r')\n",
    "compare = {}\n",
    "print('#Gamak Pond Sequencing Results', '\\n')\n",
    "\n",
    "for tax in taxonomy:\n",
    "    confirm = tax.split()\n",
    "    compare[confirm[0]] = tax\n",
    "    \n",
    "for line1 in gamakseq:\n",
    "    if line1.startswith('#'):\n",
    "        stack1 = 1\n",
    "        continue\n",
    "    else:\n",
    "        if stack1 == 1:\n",
    "            FirstLine = line1.split()\n",
    "            query1 = FirstLine[0]\n",
    "            subjt1 = FirstLine[1]\n",
    "            ident1 = FirstLine[2]\n",
    "            stack1 = 0\n",
    "            print(' Sequence: ', query1, '\\n', 'Accuracy: ', ident1, '%')\n",
    "            \n",
    "            species = compare[subjt1].split()\n",
    "            number = species[0]\n",
    "            k_species = species[1]\n",
    "            p_species = species[2]\n",
    "            c_species = species[3]\n",
    "            o_species = species[4]\n",
    "            f_species = species[5]\n",
    "            g_species = species[6]\n",
    "            s_species = species[7]\n",
    "            \n",
    "            print(' Classify: ', number)\n",
    "            print('             ', k_species)\n",
    "            print('             ', p_species)\n",
    "            print('             ', c_species)\n",
    "            print('             ', o_species)\n",
    "            print('             ', f_species)\n",
    "            print('             ', g_species)\n",
    "            print('             ', s_species, '\\n')\n",
    "            \n",
    "\n",
    "print('\\n', '\\n', '#Human and Fly ensulin compare', '\\n')\n",
    "for line2 in humanflyworm:\n",
    "    if line2.startswith('#'):\n",
    "        stack2 = 1\n",
    "        continue\n",
    "    else:\n",
    "        if stack2 == 1:\n",
    "            SecondLine = line2.split()\n",
    "            query2 = SecondLine[0]\n",
    "            subjt2 = SecondLine[1]\n",
    "            ident2 = SecondLine[2]\n",
    "            alilen = SecondLine[3]\n",
    "            mismch = SecondLine[4]\n",
    "            gapopn = SecondLine[5]\n",
    "            qstart = SecondLine[6]\n",
    "            qend = SecondLine[7]\n",
    "            sstart = SecondLine[8]\n",
    "            send = SecondLine[9]\n",
    "            evalue = SecondLine[10]\n",
    "            bscore = SecondLine[11]\n",
    "            stack2 = 0\n",
    "            print('Query Acc.ver:        ', query2)\n",
    "            print('Subject Acc.ver:      ', subjt2)\n",
    "            print('% Identity:           ', ident2)\n",
    "            print('Alignment Length:     ', alilen)\n",
    "            print('Number of Mismatches: ', mismch)\n",
    "            print('Number of Gap-opens:  ', gapopn)\n",
    "            print('Query Start:          ', qstart)\n",
    "            print('Query End:            ', qend)\n",
    "            print('Subject Start:        ', sstart)\n",
    "            print('Subject End:          ', send)\n",
    "            print('E-value:              ', evalue)\n",
    "            print('Bit Score:            ', bscore, '\\n')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
